TradingBot/Services/Strategy.cs

```diff
@@ -1,33 +1,188 @@
         _clock = clock;
+        _strategyParameters = strategyParameters;
     }
 
-    public async Task<IReadOnlyList<TradingAction>> GetTradingActionsAsync()
+    public async Task<IReadOnlyList<TradingAction>> GetTradingActionsAsync(CancellationToken token = default)
+    {
+        var strategyParameters = await _strategyParameters.GetConfigurationAsync();
+        var actions = await DetermineTradingActionsAsync(strategyParameters.MaxStocksBuyCount, strategyParameters.MinDaysDecreasing, strategyParameters.MinDaysIncreasing, strategyParameters.TopGrowingSymbolsBuyRatio, token);
+
+        return actions;
+    }
+
+    private async Task<IReadOnlyList<TradingAction>> DetermineTradingActionsAsync(int maxBuyCount, int minDaysDecreasing, int minDaysIncreasing, decimal topGrowingSymbolsBuyRatio, CancellationToken token = default)
     {
         var predictions = await _predictor.GetPredictionsAsync();
         var assets = await _assetsDataSource.GetAssetsAsync();
-        return new[]
+        var tradingActions = new List<TradingAction>();
+
+        var sellActions = new List<TradingAction>();
+        var growthRates = new List<averageGrowthRate>();
+
+        var cashAvailable = assets.Cash.BuyingPower;
+
+        foreach (var prediction in predictions)
         {
-            TradingAction.MarketBuy(new TradingSymbol("TSLA"), 0.5m, _clock.UtcNow)
-        };
+            var symbol = prediction.Key;
+            var closingPrices = new List<decimal>();
+            var currentPrice = await GetCurrentPrice(symbol, token);
+            if (currentPrice != null)
+                closingPrices.Add(currentPrice.Close);
+
+            closingPrices.AddRange(prediction.Value.Prices.Select(dailyPrice => dailyPrice.ClosingPrice).ToList());    
+
+            if (IsPriceDecreasing(closingPrices, minDaysDecreasing) && assets.Positions.TryGetValue(symbol, out var position))
+            {
+                if (position.AvailableQuantity > 0)
+                    sellActions.Add(TradingAction.MarketSell(symbol, position.AvailableQuantity, _clock.UtcNow));
+            }
+            else if (IsPriceIncreasing(closingPrices, minDaysIncreasing))
+            {
+                decimal growthRate = CalculateAverageGrowthRate(closingPrices);
+                var lowPrice0 = prediction.Value.Prices.Select(dailyPrice => dailyPrice.LowPrice).ToList()[0];
+                var lowPrice1 = prediction.Value.Prices.Select(dailyPrice => dailyPrice.LowPrice).ToList()[1];
+                var buyLimitPrice = lowPrice0 + (lowPrice1 - lowPrice0) / 2;
+                if(currentPrice != null)
+                {
+                    var tomorrowLowPrice = lowPrice0;
+                    if (tomorrowLowPrice > currentPrice.Low)
+                        buyLimitPrice = currentPrice.Low + (tomorrowLowPrice - currentPrice.Low) / 2;
+                    else
+                        buyLimitPrice = tomorrowLowPrice + (currentPrice.Low - tomorrowLowPrice) / 2;
```

Comment:
Both of these are equal to `(currentPrice.Low + tomorrowLowPrice) / 2`. 

Also I'd extract calculations like these (pretty much everything in this `else if` block) to a separate method to be more clear about what's happening here, but that's not too important right now.